src/main/java/org/jabref/preferences/JabRefPreferences.java

```diff
@@ -2067,7 +2067,8 @@ public AppearancePreferences getAppearancePreferences() {
         appearancePreferences = new AppearancePreferences(
                 getBoolean(OVERRIDE_DEFAULT_FONT_SIZE),
                 getInt(MAIN_FONT_SIZE),
-                new Theme(get(FX_THEME))
+                new Theme(get(FX_THEME)),
+                false
```

Comment:
You will need to add a new constant for the new flag and get the value here : sth like getBoolean(AUTOMATIC_THEME_DETECTION)  and then add a new listener so that the value gets stored and read on change